matrix-sdk/src/main/java/org/matrix/androidsdk/HomeServerConnectionConfig.java

```diff
@@ -75,6 +85,24 @@ public HomeServerConnectionConfig(Uri hsUri,
                                       @Nullable Credentials credentials,
                                       List<Fingerprint> allowedFingerprints,
                                       boolean pin) {
+        this(hsUri, null, credentials, new ArrayList<Fingerprint>(), false, true);
+    }
+
+    /**
+     * @param hsUri               The URI to use to connect to the homeserver
+     * @param identityServerUri   The URI to use to manage identity
+     * @param credentials         The credentials to use, if needed. Can be null.
+     * @param allowedFingerprints If using SSL, allow server certs that match these fingerprints.
+     * @param pin                 If true only allow certs matching given fingerprints, otherwise fallback to
+     *                            standard X509 checks.
+     * @param tlsExtensions       If true TLS extensions will can be used by SSL sockets
+     */
+    public HomeServerConnectionConfig(Uri hsUri,
+                                      @Nullable Uri identityServerUri,
+                                      @Nullable Credentials credentials,
+                                      List<Fingerprint> allowedFingerprints,
+                                      boolean pin,
+                                      boolean tlsExtensions) {
```

Comment:
Why adding `tlsExtensions` as a constructor parameter and not `tlsCipherSuites` and `tlsVersions`? I think there is not need to modify the constructor, the three setters are enought

Reply:
Ok. To avoid any regression in existing code, I can set `mTlsExtensions` to `true` by default. Is this ok for you ?

Reply:
Yes, I think it was already the case in the default settings:
```
  /** A modern TLS connection with extensions like SNI and ALPN available. */
  public static final ConnectionSpec MODERN_TLS = new Builder(true)
      .cipherSuites(APPROVED_CIPHER_SUITES)
      .tlsVersions(TlsVersion.TLS_1_3, TlsVersion.TLS_1_2, TlsVersion.TLS_1_1, TlsVersion.TLS_1_0)
      .supportsTlsExtensions(true)
      .build();
```